### DISCUSSION TITLE
Add APIs for saved notifications

### DISCUSSION THREAD

**Discussion by loic-sharma** [community/community#39606] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

The GitHub UI lets you [save notifications](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/viewing-and-triaging-notifications/managing-notifications-from-your-inbox#triaging-options).

However, the GitHub API does not have documented APIs for saved notifications. Below are proposed APIs to introduce this information.

See also this discussion to add APIs to mark notifications as `Done`: https://github.com/community/community/discussions/24653

## GraphQL

It seems the GraphQL API has undocumented queries and mutations to lookup/mutate notifications:

1. `notificationThreads` query
2. `markNotificationAsDone` mutation

It'd be wonderful if these were publicly available and documented :)

### Update the "List notifications for the authenticated user" API

The [List notifications for the authenticated user API](https://docs.github.com/en/rest/activity/notifications#list-notifications-for-the-authenticated-user) should be updated to include saved information.

#### Query parameters

A `saved` boolean query parameter should be added. If `true`, only shows notifications which the user has saved. Default: `false`.

#### Response

The `Thread` object should have a new `saved` property. This property should be a required boolean.

### Introduce "Mark a thread as saved" APIs

1. `POST /notifications/threads/{thread_id}/save` - Mark a notification as saved.
4. `DELETE /notifications/threads/{thread_id}/save` - Mark a notification as unsaved.

**Comment by ivan-kleshnin** [comment]
> It seems the GraphQL API has undocumented queries and mutations to lookup/mutate notifications:
> notificationThreads query
> markNotificationAsDone mutation
> It'd be wonderful if these were publicly available and documented :)

GraphQL schema does not contain such resources 🤔 
Could you please elaborate how notifications can be read from GraphQL Github API?
